snuba/migrations/snuba_migrations/querylog/0003_add_profile_fields.py

```diff
@@ -0,0 +1,134 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import (
+    Array,
+    Column,
+    LowCardinality,
+    String,
+    UInt,
+    WithDefault,
+)
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+
+
+class Migration(migration.MultiStepMigration):
+    """
+    Adds fields for query profile.
+    """
+
+    blocking = True
+
+    def __forward_migrations(self, table_name: str) -> Sequence[operations.Operation]:
+        return [
+            operations.AddColumn(
+                storage_set=StorageSetKey.QUERYLOG,
+                table_name=table_name,
+                column=Column(
+                    "clickhouse_queries.all_columns",
+                    WithDefault(
+                        Array(Array(LowCardinality(String()))),
+                        "arrayResize([['']], length(clickhouse_queries.sql))",
```

Comment:
Why do we set this default at all? Isn't this already the default for array columns that share the prefix `clickhouse_queries`?

Reply:
Because the length of all the arrays in a nested column must be the same after every insert. If it is not the insert fails.
Thus since we are creating the columns before we start populating them, they need to have a default that ensure these new columns will have the same size as the existing one (like clickhouse_queries.sql).

Reply:
Yes - but I think ClickHouse already applies this default and ensures the sizes match up, there isn't a need to specify this here.

Reply:
it did not. We created the columns in production yesterday without the default and all inserts were failing.

Reply:
Ok. Has this already been applied in prod? Does it cause the add column statement to take longer if it's actually writing these defaults into all of the existing rows as well?

Reply:
Already in prod, not aware it took longer than usual.